src/PrestaShopBundle/Translation/PrestaShopTranslatorTrait.php

```diff
@@ -53,27 +53,38 @@ public function trans($id, array $parameters = [], $domain = null, $locale = nul
             $legacy = $parameters['legacy'];
             unset($parameters['legacy']);
         }
+        $emptyParams = empty($parameters);
+        $isSprintf = $this->isSprintfString($id);
```

Comment:
```suggestion
        $isSprintf = !emptyParams && $this->isSprintfString($id);
```

I believe the optimization that @eternoendless suggested is that it's useless to call `isSprintfString` when there are no parameters since the purpose of sprintf is to replace parameters.